packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -750,6 +750,40 @@ void main() {
     expect(tester.getTopRight(find.text('text')).dx, lessThanOrEqualTo(tester.getTopLeft(find.text('s')).dx));
   });
 
+  testWidgets('InputDecorator prefixIcon/suffixIcon', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      buildInputDecorator(
+        // isEmpty: false (default)
+        // isFocused: false (default)
+        decoration: const InputDecoration(
+          prefixIcon: const Icon(Icons.pages),
+          suffixIcon: const Icon(Icons.satellite),
+          filled: true,
+        ),
+      ),
+    );
+
+    // Overall height for this InputDecorator is 48dps:
```

Comment:
Something is wrong here. The listed heights add up to 40 and the test checks for an overall height of 40, but the comment (and the point of this PR) say we're expecting an overall height of 48.

Reply:
Looking into it... might be drawing outside of bounds If I didn't correctly update something in the render object

Reply:
Confirmed locally that without a border the icons are drawn outside of bounds

Reply:
_RederDecoration.layout() should be taking the heights of the prefix and suffix icons into account when it computes containerHeight.  It doesn't appear to be doing that - which would be an existing bug.

Reply:
Not to put too fine a point on it, but. I think this:

```
    double containerHeight = contentPadding.top
      + aboveBaseline
      + belowBaseline
      + contentPadding.bottom;
```
Should be:

```
    double containerHeight = contentPadding.top
      math.max(aboveBaseline + belowBaseline, math.max(_boxSize(prefixIcon.height), _boxSize(suffixIcon.height))),
      + contentPadding.bottom;
```

Reply:
needs to take math.max over both prefix/suffix icons vs the baselines, since the don't have extra padding.